### PR TITLE
Fix product image size

### PR DESCRIPTION
**Pull request by pedromcp90**
Requested-by: [doofeeds/issues/544](https://github.com/doofinder/doofeeds/issues/544)
- Modified products and variants rest response to include a new "image_link" field

This will also require a doofeeds update to access this new field.

### CODE REVIEW

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+class REST_API_Handler
+{
+    /**
+     * Add the actions and filters needed to modify REST requests
+     *
+     * @return void
+     */
+    public static function initialize(): void
+    {
+        add_action('rest_post_dispatch', array(__CLASS__, 'add_taxonomy_image_link'), 99, 3);
```
Comment (davidmolinacano) [inline doofinder-for-woocommerce/includes/class-rest-api-handler.php:18]: Just a question: Should this be triggered if the plugin is not active?

File: doofinder-for-woocommerce/includes/class-thumbnail.php [hunk 2]
```diff
@@ -18,29 +18,14 @@ class Thumbnail {
...
 	private $post;
 
-	/**
-	 * Prepare the thumbnail size to be used when submitting
-	 * the images to the Doofinder API.
-	 *
-	 * Default WP thumbnail size (150x150) if it's available,
-	 * but if it's not we'll register our own.
-	 */
-	public static function prepare_thumbnail_size() {
```
Comment (eduardogomez97) [inline doofinder-for-woocommerce/includes/class-thumbnail.php:28]: Here you remove the prepare_thumbnail_size() function but in line [102](https://github.com/doofinder/doofinder-woocommerce/pull/174/files#diff-6de0a4988b16dc6efb250da24e08311b61d1c7c89397608a8100a58f9f5c0391R102 ) you are using it. That generates inconsistency.

**Review by eduardogomez97 — CHANGES_REQUESTED** [review]

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        if (
+            $request->get_route() === "/wc/v3/products" || //for products
+            preg_match_all('/\/wc\/v3\/products\/\d+\/variations/', $request->get_route()) //for product variations
+        ) {
+            $products = $result->data;
+            foreach ($products as $key => $product) {
+                $post = get_post($product['id']);
+                $thumbnail = new Thumbnail($post);
+                $image_link = $thumbnail->get();
+                $result->data[$key]['df_image_link'] = empty($image_link) ? wc_placeholder_img_src(Thumbnail::get_size()) : $image_link;
```
[the remark this answers is not in the document]
Comment (albertodoofinder) [reply doofinder-for-woocommerce/includes/class-rest-api-handler.php:64]: I think that can be done later, now it is better to solve the issue.

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php (hunk 1 above)
Comment (pedromcp90) [reply doofinder-for-woocommerce/includes/class-rest-api-handler.php:18]: I left it outside the WooCommerce check as it is a WordPress endpoint, but thinking it twice, it should be called only from WooCommerce clients, so it makes sense changing it to trigger only when Woocommerce is enabled.
Thank you for your appreciation.

File: doofinder-for-woocommerce/includes/class-thumbnail.php (hunk 2 above)
Comment (pedromcp90) [reply doofinder-for-woocommerce/includes/class-thumbnail.php:28]: Thanks, I've removed this function call now.

File: doofinder-for-woocommerce/includes/class-rest-api-handler.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (pedromcp90) [reply doofinder-for-woocommerce/includes/class-rest-api-handler.php:64]: OK, I'll let this for a next task

**Review by albertodoofinder — APPROVED** [review]

**Review by eduardogomez97 — APPROVED** [review]

**Review by davidmolinacano — APPROVED** [review]
:+1:

**Review by frobs — APPROVED** [review]

**Pull request merged**